### DISCUSSION TITLE
Does it work with .cshtml?

### DISCUSSION THREAD

**Discussion by devployment** [perpetualKid/GetText.NET#35] [Q&A] [answered]
Somehow I probably missed the obvious. But just tried to extract some `ICatalog` usage out of a `*.cshtml` file and the extractor seems not to find the usages. Plain `*.cs` files seem to be OK, though. 

According to this it seems to be limited to parse `*.cs` files

https://github.com/perpetualKid/GetText.NET/blob/7537c137854ab15dfbdaff12d810b9dc30870397/src/GetText.Extractor/Engine/SourceResolver/DirectorySourceResolver.cs#L24

Any pointer what it would take to support `*.cshtml` files if possible at all? Unfortunately I cannot run the project locally, as there seems to be trouble related to macOS, and my knowledge related to that general topic is quite limited. 

But for what i can "sense", `*.cshtml` might not be that far out of reach. But just guessing. Time to learn som Roslyn basics I guess. :)

**Comment by perpetualKid** [comment, marked as answer]
indeed, the extractor is currently limited to read plain cs files, as it uses Roslyn to build a syntax tree and identify string resources. Would need to see what it takes to read cshtml, but seems it would require to include the RazorTemplateEngine first to extract the code, which than could be parsed: 
https://stackoverflow.com/questions/28392086/getting-a-semanticmodel-of-a-cshtml-file

It may sound counterproductive, but a short term solution may rather be the use of the legacy GNU GetText tool (https://www.gnu.org/software/gettext/manual/gettext.html#xgettext-Invocation) which just uses plain regex - however that doesn't support i.e. C# string interpolation ($"{}")

I'll take a look but doubt to have a solution soon.

**Comment by devployment** [comment]
Thanks a lot for the fast explanation. 

> but a short term solution may rather be the use of the legacy GNU GetText tool

That's where I started looking around. But you're right, it probably will be good enough to start with. 

> I'll take a look but doubt to have a solution soon.

Not expecting that, but thank you so much for looking into that. Will setup a Windows VM to experiment around with that as well.

**Comment by perpetualKid** [comment]
as recently discovered by someone through this PR #36, seems Roslyn is able to just load Razor and cshtml files for analysis we well! There is work in progress to get this change merged, so should be available shortly. Sorry for never really been testing this before but just following (obviously outdated) information/documentation...

**Reply by perpetualKid** [reply]
implemented in f4bd222
